primitives/src/lib.rs

```diff
@@ -477,17 +477,29 @@ create_currency_id! {
     }
 }
 
+#[derive(Encode, Decode, Eq, Hash, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
+pub enum LpToken {
+    Token(TokenSymbol),
+    ForeignAsset(ForeignAssetId),
+    StableLpToken(StablePoolId),
```

Comment:
Is there a usecase for this?

Reply:
Following the interface adopted by Acala: https://github.com/AcalaNetwork/Acala/blob/1d7852f92a5e1116d8f1ad0e3d0d9e5e2d59c230/primitives/src/currency.rs#L460